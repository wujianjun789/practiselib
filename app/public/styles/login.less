@min-width: 700px;
@min-height: 630px;
@header-height:160px;
@container-mid-height:460px;

html,
body {
    min-width:100%!important;
    overflow-x:hidden;
}

.icon-orig(@image) {
    background: @image;
    background-size: 100% auto;
}
.container-login{
    position: relative;
    min-width: @min-width;
    min-height: @min-height;
    width: 100%;
    height: 100%;

    header{
        height: @header-height;
        padding-top:60px;
        position: relative;

        .main-title{
            float: left;
            h3{
                margin:0 24px 0 24px;
                font-size: 60px;
                color:#0453a5;
                line-height: 80px;
                float:right;
            }
            .login_logo{
                .icon-orig(url(/images/login_logo.png)no-repeat center);
                width:81px;
                margin-left:24px;
                height: 81px;
                float: left;
            }
        }

        .sub-title{
            border-left: 1px solid #959595;
            padding-left:20px;
            font-size: 18px;
            color: #959595;
            bottom:0;
            float: left;

            p{
                line-height: 40px;
                margin-bottom: 0px;
            }
        }
    }

    .container-mid{
        height: @container-mid-height;
        position: relative;
        .bg-cover{
            opacity:1;    
            .icon-orig(url(/images/login_background.jpg));
            height: @container-mid-height;
            width:100%;
            position: absolute;
            background-size: cover;
            background-position: 50% 50%;
            z-index:-1;
        }
        .bg-anm{
            background:rgb(24,133,255);width:100%;height: @container-mid-height;
            position: absolute;
            z-index:-2;
        }
        .login-right{
            width:370px;
            height: 290px;
            margin-top:85px;
            margin-right:24%;
            background: white;
            border-radius: 5px;
            padding: 10px 30px;

            p{
                text-align: center;

                &:first-child{
                    font-size: 20px;
                    color: #00C6FF;
                    line-height: 60px;
                    font-weight: bold;
                }

                &:last-of-type{
                    color:red;
                }
            }

            .form-group{
                margin-bottom: 20px;

                .login_user{
                    .icon-orig(url(/images/login_user.png));
                    width:17px;
                    height: 18px;
                    margin: 8px;
                }

                .login_password{
                    .icon-orig(url(/images/login_password.png));
                    width:17px;
                    height: 20px;
                    margin: 7px 8px;
                }

            }

            .btn-login{
                background: -webkit-gradient(linear, left 0, right 0, from#00bcff, to #00d8ff)!important;   /*  for safari */
                background-image: linear-gradient(to right, #00bcff, #00d8ff)!important;
                color: #fff!important;
                font-size: 18px;
                border-radius:20px;
                margin-top:12px;
                box-shadow: 0 2px 7px 1px rgba(0,175,255,0.35);
                // outline: none!important;
                &:hover {
                    background: -webkit-gradient(linear, left 0, right 0, from#00b1f0, to #00cbf0)!important;   /*  for safari */
                    background-image: linear-gradient(to right, #00b1f0, #00cbf0)!important;
                }
            }
        }
    }

    footer{
        position: absolute;
        width: 100%;
        text-align: center;
        height: 35px; 
        color: #959595;
        line-height: 35px;
        margin-top:20px;
    }
}

