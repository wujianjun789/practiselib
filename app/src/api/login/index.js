import {login} from '../../util/network';

export function loginHandler(username, password,cbSuccess,cbFail) {
    if(username.length <= 2 || password.length <= 4){
        cbFail && cbFail();
        return;
    }

    login({name:username, password:password}, response=>{
        cbSuccess && cbSuccess();
    }, err=>{
        cbFail && cbFail();
    })
        // cbSuccess && cbSuccess();
    
}