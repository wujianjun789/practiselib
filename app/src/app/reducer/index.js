/**
 * create by Azrael on 2017/7/4
 */

import {
    MODULE_INIT
} from '../actionType/index';

export const initialState = {
    title: "StarRiver",
    name: "智慧路灯管理系统",
    items: [
        // {key: 'asset', title: '资产管理', link: '/assetManage/manage'},
        // {key: 'permission', title: '权限管理', link: '/permissionManage'},
        // {key: 'maintenance', title: '系统运维', link: '/systemOperation/lampConCenter'},
        // {key: 'control', title: '系统控制', link: '/'},
        // {key: 'report', title: '报表管理', link: '/'},
        // {key: 'publish', title: '媒体发布', link: '/'},
        // {key: 'visual', title: '可视化', link: '/'},
        // {key: 'domain', title: '域管理', link: '/domainManage/domainEdit'},
    ],
};

export default function app(state=initialState, action) {
    switch(action.type) {
        case MODULE_INIT:
            return moduleInit(state, action.data);
        default:
            return state;
    }
}

function moduleInit(state, data) {
    data = data.map((item, index) => {
        switch(item.key) {
            case 'asset':
                item.title = '资产管理';
                item.link = '/assetManage/manage';
                break;
            case 'permission':
                item.title = '权限管理';
                item.link = '/permissionManage';
                break;
            case 'maintenance':
                item.title = '系统运维';
                item.link = '/systemOperation/config/lc';
                break;
            case 'control':
                item.title = '系统控制';
                item.link = '/';
                break;
            case 'report':
                item.title = '报表管理';
                item.link = '/';
                break;
            case 'publish':
                item.title = '媒体发布';
                item.link = '/';
                break;
            case 'asset':
                item.title = '可视化';
                item.link = '/';
                break;
            case 'domain':
                item.title = '域管理';
                item.link = '/domainManage/domainEdit';
                break;
        }
        return item;
    });
    return Object.assign({}, state, {items: data});
}