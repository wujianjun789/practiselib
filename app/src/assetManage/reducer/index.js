/**
 * Created by a on 2017/4/21.
 */

import Immutable from 'immutable';
const initialState = {
};


export default function login (state=Immutable.fromJS(initialState), action) {
    switch(action.type) {
        default:
            return state;
    }
}