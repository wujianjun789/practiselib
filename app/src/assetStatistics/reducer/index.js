/**
 * Created by a on 2017/4/21.
 */
import {
    ON_CHANGE
} from '../actionType/index'
import Immutable from 'immutable';
const initialState = {
    data: [
        {domain:"闵行区", deviceName:"灯集中控制器", soft_v:"1.0", sys_v:"1.0", core_v:"1.0", har_v:"1.0",
        vendor_info:"上海三思", con_type:485},
        {domain:"闵行区", deviceName:"灯集中控制器", soft_v:"1.0", sys_v:"1.0", core_v:"1.0", har_v:"1.0",
        vendor_info:"上海三思", con_type:485}
    ],
    domain:{list:[{id:1, value:'域'},{id:2, value:'域2'}], value:'域'},
    device:{list:[{id:1, value:'灯集中控制器'},{id:2, value:'集中控制'}], value:'灯集中控制器'},
    search:{placeholder:'输入素材名称', value:''}
};


export default function assetStatistics (state=Immutable.fromJS(initialState), action) {
    switch(action.type) {
        case ON_CHANGE:
            if(action.id == 'search'){
                return state.updateIn([action.id, 'value'], (v)=>action.data);
            }

            return state.updateIn([action.id, 'value'], (v)=>{
                return state.getIn([action.id, 'list', action.data, 'value'])
            })
            break;
        default:
            return state;
    }
}