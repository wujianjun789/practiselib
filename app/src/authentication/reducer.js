import {deleteCookie} from '../util/cache'
import { getAuth } from './auth';
export const initialState = { auth: getAuth() };
export default function auth(state = initialState, action) {
  switch (action.type) {
    case 'LOGED_IN':
      return { auth: getAuth() };
    case 'LOGED_OUT':
      deleteCookie('user');
      return { auth: getAuth() };
    default:
      return state;
  }
}