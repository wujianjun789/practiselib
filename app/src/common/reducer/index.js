
import { combineReducers } from 'redux';
import { routerReducer as routing } from 'react-router-redux';

import app from '../../App/reducer/index';
import overlayer from './overlayer';

// import login from '../../login/reducer/index'
import assetManage from '../../assetManage/reducer/index'
import assetStatistics from '../../assetStatistics/reducer/index'
import domainManage from '../../domainManage/reducer/index'
import treeView from './treeView';
import userCenter from './userCenter';
import permissionManage from '../../permissionManage/reducer/index'
import sysOperation from '../../systemOperation/reducer/index';
import lampConCenter from '../../systemOperation/reducer/lampConCenter';
const reducer = combineReducers({
    routing,
    overlayer,

    userCenter,
    treeView,
    // login,
    app,
    assetManage,
    assetStatistics,

    domainManage,
    permissionManage,

    sysOperation,
    lampConCenter
});
export default reducer;
