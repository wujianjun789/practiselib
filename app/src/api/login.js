/**
 * Created by a on 2017/7/26.
 */
import {LOGED_IN} from '../authentication/actionTypes'
import {login} from '../util/network';
import {setAuth} from '../authentication/auth'
import {HOST_IP, getHttpHeader, httpRequest} from '../util/network'

export const loginHandler = (username, password,cbSuccess,cbFail) => dispatch => {
    if(username.length <= 2 || password.length <= 4){
        cbFail && cbFail();
        return;
    }

    login({username:username, password:password}, response=>{
        let headers = getHttpHeader();
        httpRequest(`${HOST_IP}/users/${response.userId}`, {
            method: 'GET',
            headers: headers,
          }, res=>{
            response.roleId = res.roleId;
            setAuth(response);
            dispatch({ type: LOGED_IN });
            cbSuccess && cbSuccess();
          })
        
    }, err=>{
        cbFail && cbFail();
    })

}