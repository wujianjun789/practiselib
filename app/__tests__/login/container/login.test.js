import React from 'react';
import {mount} from 'enzyme';
import {Provider} from 'react-redux';
import configureStore from '../../../src/store/configureStore';
import Login from '../../../src/login/container/Login';
import {initialState as state} from '../../../src/login/reducer';

describe('<login /> HOC', () => {
    const store = configureStore();
    it('render normal', () => {
        const root = mount(<Provider store={store}>
            <Login />
        </Provider>);

        const login = root.find('Login');
        expect(login.length).toBe(1);
        expect(login.prop('data')).toBe(state);
    });
})