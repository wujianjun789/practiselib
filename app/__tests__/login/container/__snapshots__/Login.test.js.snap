exports[`<Login /> snapshot 1`] = `
<div
  className="container-login"
  onKeyDown={[Function]}>
  <header>
    <div
      className="main-title">
      <span
        className="login_logo" />
      <h3>
        StarRiver
      </h3>
    </div>
    <div
      className="sub-title">
      <p>
        智慧路灯管理系统
      </p>
      <p>
        Smart Pole Control System
      </p>
    </div>
  </header>
  <div
    className="container-mid">
    <div
      className="bg-cover" />
    <div
      className="login-right pull-right">
      <p>
        用户登录
      </p>
      <div
        className="form-group has-feedback">
        <input
          className="form-control"
          id="username"
          onChange={[Function]}
          onFocus={[Function]}
          type="text"
          value="" />
        <span
          className="login_user form-control-feedback" />
      </div>
      <div
        className="form-group has-feedback">
        <input
          className="form-control"
          id="password"
          onChange={[Function]}
          onFocus={[Function]}
          type="password"
          value="" />
        <span
          className="login_password form-control-feedback" />
      </div>
      <p
        style={
          Object {
            "visibility": "hidden",
          }
        }>
        用户名或密码错误
      </p>
      <button
        className="btn btn-block btn-login"
        onClick={[Function]}
        type="button">
        登录
      </button>
    </div>
  </div>
  <footer>
    Copyright © 2017 SANSITECH.All rights reserverd.
  </footer>
</div>
`;
